src/pages/Bed/Bed.js

```diff
@@ -1,14 +1,28 @@
 import React from "react"
+import { useQuery } from "@apollo/client"
+
+import { SINGLE_BED_QUERY } from "queries"
 import { useUrlParam } from "hooks"
 import { BedBuilder } from "./BedBuilder"
 
 export function Bed() {
-  const bedName = useUrlParam("name")
+  const id = useUrlParam("id")
+
+  const { data, loading, error } = useQuery(SINGLE_BED_QUERY, {
+    variables: { id: parseInt(id) },
+  })
 
+  if (loading) return <p>Loading...</p>
+  if (error) return <p>{error.message}</p>
+  const { bed } = data
   return (
     <>
-      <h2>{bedName}</h2>
-      <BedBuilder />
+      <h2>{data.bedName}</h2>
+      <BedBuilder
+        height={bed.height}
+        width={bed.width}
```

Comment:
Can you please change "height" to "length"? I know that I introduced the offending term but we should prob excise it at (this) the first opportunity.

Reply:
What do you think about merging this for now, since it's functional? I can make my next task to clean this up but it will probably take some time since I set it up the same way on the backend and will need to fix that as well.

Reply:
I was actually able to work this out quicker than expected!